lms/djangoapps/instructor/cybersource_enrollment_report.py

```diff
@@ -0,0 +1,130 @@
+                course_enrollment=course_enrollment
+            )
+
+            # from where the user get here
+            if registration_code_redemption is not None:
+                enrollment_source = _('Used Registration Code')
+            elif paid_course_reg_item is not None:
+                enrollment_source = _('Credit Card - Individual')
+            else:
+                enrollment_source = _('Manually Enrolled')
+
+        enrollment_date = getattr(course_enrollment, 'created').strftime("%B %d, %Y")
+        currently_enrolled = getattr(course_enrollment, 'is_active')
+
+        course_enrollment_data = [enrollment_date, currently_enrolled, enrollment_source, enrollment_role]
+        return course_enrollment_data
+
+    def get_payment_info(self, user, course_id):
+        """
+        Returns the User Payment information.
+        """
+        course_enrollment = CourseEnrollment.get_enrollment(user=user, course_key=course_id)
+        paid_course_reg_item = PaidCourseRegistration.get_course_item_for_user_enrollment(
+            user=user,
+            course_id=course_id,
+            course_enrollment=course_enrollment
+        )
+        # check if the user made a single self purchase scenario
+        # for enrollment in the course.
+        if paid_course_reg_item is not None:
+            coupon_redemption = CouponRedemption.objects.select_related('coupon').filter(
+                order_id=paid_course_reg_item.order_id)
+            coupon_codes = [redemption.coupon.code for redemption in coupon_redemption]
+            coupon_codes = ", ".join(coupon_codes)
+            payment_data = [
+                paid_course_reg_item.unit_cost, paid_course_reg_item.line_cost, coupon_codes,
+                paid_course_reg_item.status, paid_course_reg_item.order_id
+            ]
+        else:
+            # check if the user used a registration code for the enrollment.
+            registration_code_redemption = RegistrationCodeRedemption.registration_code_used_for_enrollment(
+                course_enrollment)
+            if registration_code_redemption is not None:
+                registration_code = registration_code_redemption.registration_code
+                if getattr(registration_code, 'invoice_item_id'):
+                    list_price = getattr(registration_code.invoice_item, 'unit_price')
+                    total_amount = registration_code_redemption.registration_code.invoice.total_amount
+                    qty = registration_code_redemption.registration_code.invoice_item.qty
+                    payment_amount = total_amount / qty
+                    is_valid = registration_code_redemption.registration_code.invoice.is_valid
+                    transaction_reference_number = registration_code_redemption.registration_code.invoice_id
+                    if is_valid:
+                        status = _('Invoice Paid')
+                    else:
+                        status = _('Invoice Outstanding')
```

Comment:
There are some new tables to track payments against Invoices that @wedaly 's team added (maybe about 2 months ago). I think they will be helpful here. I believe it is called InvoiceTransaction.